common/src/main/scala/org/apache/celeborn/common/util/Utils.scala

```diff
@@ -396,22 +396,14 @@ object Utils extends Logging {
 
   private var customHostname: Option[String] = sys.env.get("CELEBORN_LOCAL_HOSTNAME")
 
+  // for testing
   def setCustomHostname(hostname: String) {
-    // DEBUG code
     Utils.checkHost(hostname)
     customHostname = Some(hostname)
   }
 
-  def localCanonicalHostName: String = {
-    customHostname.getOrElse(localIpAddress.getCanonicalHostName)
-  }
-
-  def localHostName: String = {
-    customHostname.getOrElse(localIpAddress.getHostAddress)
-  }
-
-  def localHostNameForURI: String = {
-    customHostname.getOrElse(InetAddresses.toUriString(localIpAddress))
+  def localHostName(conf: CelebornConf): String = customHostname.getOrElse {
+    if (conf.bindPreferIP) localIpAddress.getHostAddress else localIpAddress.getCanonicalHostName
```

Comment:
In K8s `localIpAddress.getCanonicalHostName` is the FQDN, same as result of `hostname -f`